### PR TITLE
Better Engineering Priority Rewards

### PR DESCRIPTION
**Pull request by DorainGrey**
## What this does
The Engineering Priority rewards have been changed to the following.

Chief Engineer: Omnitool, Atmospheric Analysis Scope, Poly's Cracker & can of Supermatter Sea Salt Soda
Station Engineer: Omnitool & can of Supermatter Sea Salt Soda
Atmospheric Technician: Atmospheric Analysis Scope & can of Energizer
Mechanic:  Valuable Asteroid & can of Energizer 

Adds two drinks called Energizer & Supermatter Sea Salt Soda, contains Hyronalin & cures hallucinations. Credit to @tinysaturn for sprites.
Also adds valuable asteroid for priority spawn mechanics breaks down into 5 sheets of Uranium, Silver, Gold & Plasma. Tech level engineering 3

image

## Why it's good
Engineering Priority Rewards are terrible, they are tools you can easily obtain in your locker or Department vending machine.  Getting a more unique reward similar to how Medbay & Security do, will encourage people to join when they are requested.

## Changelog
:cl:
 * rscadd: Added Energizer & Supermatter Sea Salt Soda cans, they contains Hyronalin & cures hallucinations.
 * rscadd: Valuable Asteroid, priority reward for Mechanics, breaks down into 5 sheets of Uranium, Silver, Gold & Plasma.
 * tweak: Changed Chief Engineer Priority Reward to an Omnitool, Atmospheric Analysis Scope & can of Supermatter Sea Salt Soda
 * tweak: Changed Station Engineer Priority Reward to an Omnitool & can of Supermatter Sea Salt Soda
 * tweak: Changed Atmospheric Technician Priority Reward to an Atmospheric Analysis Scope & can of Energizer
 * tweak: Changed Mechanic Priority Reward to an Valuable Asteroid & can of Energizer

### CODE REVIEW

**Comment by pedr69** [comment]
hyronalin OD does nothing, right?
its metabolism is 0.05, drinking a sip of the coffee will give you 12.5u hyro, which will be in you for 250 ticks or ~500 seconds

**Comment by wotisjanitor** [comment]
this is super gay

**Comment by wotisjanitor** [comment]
oh nvm misread, carry on

**Comment by DorainGrey** [comment]
> hyronalin OD does nothing, right? its metabolism is 0.05, drinking a sip of the coffee will give you 12.5u hyro, which will be in you for 250 ticks or ~500 seconds

I chugged a full can and got no tox damage on analyzer, so I assume it's fine. It's got default overdose at 30 units. I'll reduce it 0.1 just in case.

**Comment by Eneocho** [comment]
The energizer should give you some Sports Drink too, since it Energizes (you) as well. Chug the can then run on the treadmill like a madman.

**Comment by Kurfursten** [comment]
Advanced scanner is a weird reward. It lets you scan machines and computers your ID does not let you access (different from the hacked syndie device scanner). How often does this come up? It's mostly if you wanted to flatpack the robotics console. Very rare application.

**Comment by pedr69** [comment]
why would you ever scan the robotics console anyway?
the spacepod bay console is public

**Comment by Kurfursten** [comment]
I am talking about the computer that allows you to blow up the cyborgs, the "robotics control console"

**Comment by SonixApache** [comment]
that can is super cute well done 👍

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -8806,6 +8806,20 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
 		if(H.job in list("Security Officer", "Head of Security", "Detective", "Warden"))
 			H.heal_organ_damage(1, 1) //liquid sprinkles!
 
+
+/datum/reagent/drink/coffee/engicoffee
+	name = "NT Standard Battery Acid"
+	id = ENGICOFFEE
+	description = "This Plasma Infused Brew, will fix what ails you."
+	mug_icon_state = "engicoffeee"
+	mug_name = "\improper Wake-Up Call"
```
Comment (SonixApache) [inline code/modules/reagents/Chemistry-Reagents.dm]: you missed this one

**Comment by tinysaturn** [comment]
> Advanced scanner is a weird reward. It lets you scan machines and computers your ID does not let you access (different from the hacked syndie device scanner). How often does this come up? It's mostly if you wanted to flatpack the robotics console. Very rare application.

The advanced analyser also holds 20 designs in its memory which is pretty nice

**Comment by DorainGrey** [comment]
> Advanced scanner is a weird reward. It lets you scan machines and computers your ID does not let you access (different from the hacked syndie device scanner). How often does this come up? It's mostly if you wanted to flatpack the robotics console. Very rare application.

AFAIK, this isn't actually used. Otherwise I would have chosen a different item, the code is there to do all the checks but the MECH_SCAN_ACCESS flag is not assigned to any machines.

If this were to change or anyone has a better idea of a reward for mechanics, I can change it. Not really a core part of the PR, just feel kind of guilty leaving mechanics with their terrible reward of gloves.

**Comment by Kurfursten** [comment]
Extra assembly frame? Extra spacepod core?

I also considered spawning them in with An Actual Asteroid
> Takes up 2 hands and is a huge object
> Heavy blunt weapon
> Recycle it at your protolathe, worth 5 sheets of uranium, silver, gold, plasma

Mechanics often get forgotten about by miners, especially if they latejoin. This could be fun.

**Comment by Eneocho** [comment]
I was gonna suggest something like that. A baggie of materials or something, but the actual asteroid idea sounds better. You could reuse the toy asteroid sprite too.

**Comment by ConfusedCactus** [comment]
I don't like giving people otherwise trader-exclusive stuff.

**Comment by wotisjanitor** [comment]
> I don't like giving people otherwise trader-exclusive stuff.

🤓

**Comment by Kurfursten** [comment]
I understand your concern cactus, and I share it. I wouldn't want it to set a major precedent, but I'm also not very worried in this case.

Omnitools are cool but not the most valuable item in the crate, and latejoin rewards are pretty unreliable in terms of acquisition. I would keep an eye on creeping trader items to the main station but this time I think it's okay.

**Comment by Eneocho** [comment]
The engineer could get a FRASS sphere (Hive away mission), the single use anti-shard spheres (the traders have the SASS multi use ones). You'd still want the trader's multi-use sphere but can survive one (1) engineering moment.

**Comment by Kurfursten** [comment]
SASS is the original and FRASS is the trader one, but point taken

**Comment by SECBATON-GRIFFON** [comment]
don't really like just giving trader gear out for this (unless someone can prove other jobs already get it)
CE should get the advanced tool belt loaded with stuff
don't like mechanics getting the CE device analyser (give them paper blueprints for some vital station stuff instead)
rest is fine

**Comment by Kurfursten** [comment]
Other jobs don't get it. I think it's a fair objection since it might open the door to other creep.

**Comment by Eneocho** [comment]
>CE should get the advanced tool belt loaded with stuff 

The advanced toolbelt spawns in the locker and it takes you around 20 seconds to move everything from your regular belt to the advanced one and insert the special tools (screwsolder, blueprint, RCD, RPD and inflatable deployer).

**Comment by DorainGrey** [comment]
This has been updated to reflect feedback.
It is now ready for merging @Kurfursten

**Comment by DorainGrey** [comment]
> don't really like just giving trader gear out for this (unless someone can prove other jobs already get it) CE should get the advanced tool belt loaded with stuff don't like mechanics getting the CE device analyser (give them paper blueprints for some vital station stuff instead) rest is fine

This has been changed, mechanics now get valuable asteroid instead.
Regarding omnitool, it's a niche but handy item. Adding it to latejoin rewards isn't going to unbalance the station or make it widely available.

**Pull request merged**